Quaver.Shared/Screens/Edit/Input/EditorInputManager.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Quaver.Shared.Graphics.Notifications;
+using Quaver.Shared.Screens.Edit.Dialogs;
+using Quaver.Shared.Screens.Edit.Plugins;
+using Quaver.Shared.Screens.Edit.UI;
+using Quaver.Shared.Screens.Edit.UI.Playfield.Waveform;
+using Wobble;
+using Wobble.Graphics;
+using Wobble.Graphics.UI.Dialogs;
+using Wobble.Input;
+
+namespace Quaver.Shared.Screens.Edit.Input
+{
+    public class EditorInputManager
+    {
+        public EditorInputConfig InputConfig { get; }
+        public EditScreen Screen { get; }
+        private EditScreenView View { get; }
+
+        private Dictionary<Keybind, HashSet<KeybindActions>> KeybindDictionary;
+        private GenericKeyState PreviousKeyState;
+
+        private const int HoldRepeatActionDelay = 250;
+        private const int HoldRepeatActionInterval = 25;
+        private Dictionary<KeybindActions, long> LastActionPress = new Dictionary<KeybindActions, long>();
+        private Dictionary<KeybindActions, long> LastActionTime = new Dictionary<KeybindActions, long>();
+
+        private static HashSet<KeybindActions> HoldRepeatActions = new HashSet<KeybindActions>()
+        {
+        };
+
+        private static HashSet<KeybindActions> HoldAndReleaseActions = new HashSet<KeybindActions>()
+        {
+        };
+
+        private static HashSet<KeybindActions> EnabledActionsDuringGameplayPreview = new HashSet<KeybindActions>()
+        {
+        };
+
+        public EditorInputManager(EditScreen screen)
+        {
+            InputConfig = EditorInputConfig.LoadFromConfig();
+            KeybindDictionary = InputConfig.ReverseDictionary();
+            PreviousKeyState = new GenericKeyState(GenericKeyManager.GetPressedKeys());
+            Screen = screen;
+            View = (EditScreenView)screen.View;
+        }
+
+        public void HandleInput()
+        {
+            if (DialogManager.Dialogs.Count != 0) return;
+            if (View.IsImGuiHovered) return;
+
+            HandleKeypresses();
+            HandleKeyReleasesAfterHoldAction();
+            HandlePluginKeypresses();
+
+            HandleMouseInputs();
+        }
+
+        private void HandleKeypresses()
```

Comment:
```suggestion
        private void HandleKeyPresses()
```

And all other instances of Presses not being capitalized.